docs-go/dotprompt.md

```diff
@@ -0,0 +1,317 @@
+# Managing prompts with Dotprompt
+
+Firebase Genkit provides the Dotprompt plugin and text format to help you write
+and organize your generative AI prompts.
+
+Dotprompt is designed around the premise that _prompts are code_. You write and
+maintain your prompts in specially-formatted files called dotprompt files, track
+changes to them using the same version control system that you use for your
+code, and you deploy them along with the code that calls your generative AI
+models.
+
+To use Dotprompt, first create a `prompts` directory in your project root and
+then create a `.prompt` file in that directory. Here's a simple example you
+might call `greeting.prompt`:
+
+```none
+---
+model: vertexai/gemini-1.5-pro
+config:
+  temperature: 0.9
+input:
+  schema:
+    location: string
+    style?: string
+    name?: string
+  default:
+    location: a restaurant
+---
+
+You are the world's most welcoming AI assistant and are currently working at {{location}}.
+
+Greet a guest{{#if name}} named {{name}}{{/if}}{{#if style}} in the style of {{style}}{{/if}}.
+```
+
+To use this prompt:
+
+- {Go}
+
+  Install the `dotprompt` plugin:
+
+  ```posix-terminal
+  go get github.com/firebase/genkit/go/plugins/dotprompt
+  ```
+
+  Then, load the prompt using `Open`:
+
+  ```go
+  import "github.com/firebase/genkit/go/plugins/dotprompt"
+  ```
+
+  ```go
+  dotprompt.SetDirectory("prompts")
+  prompt, err := dotprompt.Open("greeting")
+  ```
+
+  You can call the prompt's `Generate` method to render the template and pass it
+  to the model API in one step:
+
+  ```go
+  // The .prompt file specifies vertexai/gemini-1.5-pro, so make sure it's set
+  // up:
+  projectID := os.Getenv("GCLOUD_PROJECT")
+  vertexai.Init(context.Background(), projectID, "us-central1")
```

Comment:
Better to write

    ctx := context.Background()

and then use `ctx` here and below rather than calling `context.Background` multiple times.